verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -42,25 +51,177 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
     return status;
   }
 
-  // TODO(fangism): parse a subcommand-specific flag for replacement option
-  // See documentation for 'replacement' arg of the library function:
-  verilog::StripVerilogComments(source_contents, &outs, ' ');
+  char replace_char;
+  if (args.size() == 1) {
+    replace_char = ' ';
+  } else if (args.size() == 2) {
+    absl::string_view replace_str(args[1]);
+    if (replace_str.empty())
+      replace_char = '\0';
+    else if (replace_str.length() == 1)
+      replace_char = replace_str[0];
+    else
+      return absl::InvalidArgumentError(
+          "Replacement must be a single character.");
+  } else {
+    return absl::InvalidArgumentError("Too many arguments.");
+  }
+
+  verilog::StripVerilogComments(source_contents, &outs, replace_char);
 
   return absl::OkStatus();
 }
 
-static const std::pair<absl::string_view, SubcommandEntry> kCommands[] = {
-    {"strip-comments",  //
-     {&StripComments,   //
-      R"(strip-comments file
+static absl::Status PreprocessSingleFile(const char* source_file, std::istream&,
+                                         std::ostream& outs, std::ostream&) {
+  std::string source_contents;
+  if (auto status = verible::file::GetContents(source_file, &source_contents);
+      !status.ok()) {
+    std::cerr << "ERROR: passed file can't be open\n.";
+    return status;
+  }
+  verilog::VerilogPreprocess::Config config;
+  config.filter_branches = 1;
+  // config.expand_macros=1;
+  verilog::VerilogPreprocess preprocessor(config);
+  verilog::VerilogLexer lexer(source_contents);
+  verible::TokenSequence lexed_sequence;
+  for (lexer.DoNextToken(); !lexer.GetLastToken().isEOF();
+       lexer.DoNextToken()) {
+    // For now we will store the syntax tree tokens only, ignoring all the
+    // white-space characters. however that should be stored to output the
+    // source code just like it was, but with conditionals filtered.
+    if (verilog::VerilogLexer::KeepSyntaxTreeTokens(lexer.GetLastToken()))
+      lexed_sequence.push_back(lexer.GetLastToken());
+  }
+  verible::TokenStreamView lexed_streamview;
+  // Initializing the lexed token stream view.
+  InitTokenStreamView(lexed_sequence, &lexed_streamview);
+  verilog::VerilogPreprocessData preprocessed_data =
+      preprocessor.ScanStream(lexed_streamview);
+  auto& preprocessed_stream = preprocessed_data.preprocessed_token_stream;
+  for (auto u : preprocessed_stream) outs << *u << '\n';
+  // for debugging.
+  for (auto& u : preprocessed_data.errors) outs << u.error_message << '\n';
+
+  //  parsing just as a trial
```

Comment:
What are the following lines for ? Looks like an expensive string concatenation of everything, then parsing it wihtout using the result. I suspect this is some leftover debugging code ?

Reply:
Yes, I removed it.